crates/fuel-core/src/schema/tx/types.rs

```diff
@@ -1059,6 +1060,23 @@ impl DryRunTransactionExecutionStatus {
     }
 }
 
+pub struct DryRunStorageReads {
+    pub tx_statuses: Vec<DryRunTransactionExecutionStatus>,
+    pub storage_reads: Vec<StorageReadReplayEvent>,
+}
+
+#[Object]
+impl DryRunStorageReads {
+    async fn tx_statuses(&self) -> Vec<DryRunTransactionExecutionStatus> {
+        self.tx_statuses.clone()
+    }
+
+    async fn storage_reads(&self) -> Vec<StorageReadReplayEvent> {
+        self.storage_reads.clone()
+    }
+}
```

Comment:
Do we really need to clone here? Maybe it is possible to return a reference instead?

Reply:
Good catch. Some other functions here do have to clone as they wrap the result in a struct that doesn't have a reference, but in this case we can indeed just return a reference.

Reply:
1deb59dfbe